src/cmap/mod.rs

```diff
@@ -105,34 +203,21 @@ pub(crate) struct ConnectionPoolInner {
     /// them to the pool.
     min_pool_size: Option<u32>,
 
-    /// The ID of the next connection created by the pool.
-    next_connection_id: AtomicU32,
-
-    /// If a checkout operation takes longer than `wait_queue_timeout`, the pool will return an
-    /// error. If `wait_queue_timeout` is `None`, then the checkout operation will not time out.
-    wait_queue_timeout: Option<Duration>,
-
-    /// The TLS options to use for the connections. If `tls_options` is None, then TLS will not be
-    /// used to connect to the server.
-    tls_options: Option<TlsOptions>,
-
-    /// The total number of connections currently in the pool. This includes connections which are
-    /// currently checked out of the pool.
-    total_connection_count: AtomicU32,
-
-    /// Connections are checked out by concurrent threads on a first-come, first-server basis. This
-    /// is enforced by threads entering the wait queue when they first try to check out a
-    /// connection and then blocking until they are at the front of the queue.
+    /// The queue that operations wait in to check out a connection.
+    ///
+    /// A thread will only reach the front of the queue if a connection is available to be checked out
+    /// or if one could be created without going over `max_pool_size`.
     wait_queue: WaitQueue,
 }
 
 impl ConnectionPool {
```

Comment:
A lot of the changes here are simply moving the implementations to inner. Instead of having different methods implemented on the different structs, I decided it was clearer to just implement everything on the inner type, and simply call through to the methods from `ConnectionPool`. This nicely splits the "internal API" of the pool from the internal implementation of the pool itself.